geom/src/lib.rs

```diff
@@ -49,6 +49,20 @@ pub fn trim_f64(x: f64) -> f64 {
     (x * 10_000.0).round() / 10_000.0
 }
 
+/// Serializes a trimmed `f64` as an `i32` to save space.
+fn serialize_f64<S: Serializer>(x: &f64, s: S) -> Result<S::Ok, S::Error> {
+    // So a trimmed f64's range becomes 2**31 / 10,000 =~ 214,000, which is plenty
+    // We don't need to round() here; trim_f64 already handles that.
```

Comment:
> We don't need to round() here; trim_f64 already handles that.

Is there a reason to keep trim_64  as something that happens as a precondition to this method? It's not obvious to me that `trim_f64` is being called in all the places that use this method.

It's a minor point - so feel free to take it or leave it.

Reply:
To clarify my suggestion: would it make sense to delete trim_f64 and instead put that logic in this method.

Reply:
>  It's not obvious to me that trim_f64 is being called in all the places that use this method.

Unless I missed one with grep, I think all of the `trim_f64` and `serialize_f64` cases match up in `geom`. We could be more type-safe here and make `trim_f64` return some `TrimmedF64(f64)` newtype and make the serialize methods work off of that. But it feels like a bit overkill for a small API...

> would it make sense to delete trim_f64 and instead put that logic in this method.

I don't think so.. we don't want a behavioral difference in using an object that we create in-memory, vs one we load from serde. The f64 needs to be exactly the same in both cases. If we did the rounding only when serializing, then we'd get slightly different results if we
1) import a `Map`, immediately run a simulation with it
2) load a `Map` from disk, then run a simulation with it
since the rounding would only happen during serialization